src/Agoda.IoC.NetCore/StartupExtension.cs

```diff
@@ -25,6 +26,18 @@ public static class StartupExtension
             return rtn;
         }
 
+        public static IServiceProvider UseAgodaIoCStartupable(this IServiceProvider app)
+        {
+            foreach (var startupable in app.GetServices<IStartupable>())
+            {
+                if (startupable == null)
+                    throw new Exception(
+                        "No service found that inherit from IStartupable but UseAgodaIoCStartupable was called");
+                startupable.Start();
```

Comment:
maybe for a future iteration however it would be nice if you could specify the degree of parallelism.